qmail-rcptcheck.c

```diff
@@ -0,0 +1,167 @@
+#include <unistd.h>
+#include "control.h"
+#include "env.h"
+#include "stralloc.h"
+#include "substdio.h"
+#include "wait.h"
+
+static char outbuf[128];
+static struct substdio ssout = SUBSTDIO_FDBUF(write,1,outbuf,sizeof outbuf);
+static char errbuf[128];
+static struct substdio sserr = SUBSTDIO_FDBUF(write,2,errbuf,sizeof errbuf);
+
+struct rcptcheck_api {
+  char *(*get_sender)();
+  char *(*get_recipient)();
```

Comment:
The return type should be const to make it clear that the caller may not modify the value.